code/datums/status_effects/xeno_buffs.dm

```diff
@@ -24,6 +27,335 @@
+	link_owner.use_plasma(plasma_cost)
+
+/// Removes the status effect on death.
+/datum/status_effect/stacking/essence_link/proc/handle_death(datum/source)
+	SIGNAL_HANDLER
+	essence_link_action.end_ability()
+
+/// Toggles the link depending on whether the linked xenos are still in range or not.
+/datum/status_effect/stacking/essence_link/proc/handle_dist(datum/source)
+	SIGNAL_HANDLER
+	var/within_range = get_dist(link_owner, link_target) <= DRONE_ESSENCE_LINK_RANGE
+	if(within_range == was_within_range)
+		return
+	was_within_range = within_range
+	toggle_link(was_within_range)
+	link_owner.balloon_alert(link_owner, was_within_range ? ("Link reestablished") : ("Link faltering"))
+	link_target.balloon_alert(link_target, was_within_range ? ("Link reestablished") : ("Link faltering"))
+
+/// Shares the Resin Jelly buff with the linked xeno.
+/datum/status_effect/stacking/essence_link/proc/share_jelly(datum/source)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/buff_owner
+	var/mob/living/carbon/xenomorph/buff_target
+
+	if(source == link_target)
+		buff_owner = link_target
+		buff_target = link_owner
+	else
+		buff_owner = link_owner
+		buff_target = link_target
+
+	buff_owner.balloon_alert(buff_owner, "Buff shared")
+	buff_target.balloon_alert(buff_target, "Buff shared")
+	buff_target.visible_message(span_notice("[buff_target]'s chitin begins to gleam with an unseemly glow..."), \
+		span_xenonotice("Through the essence link, [buff_owner] has shared their resin jelly with you."))
+	INVOKE_ASYNC(buff_target, /mob/living/carbon/xenomorph.proc/emote, "roar")
+	buff_target.apply_status_effect(STATUS_EFFECT_RESIN_JELLY_COATING)
+
+/// Shares heals with the linked xeno.
+/datum/status_effect/stacking/essence_link/proc/share_heal(datum/source, amount)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/heal_owner
+	var/mob/living/carbon/xenomorph/heal_target
+
+	// Prevents passive regeneration from triggering this. Most values do not go past 60.
+	if(amount > -60)
+		return
+	// Prevents duplicate heals. Some heals restore both damage types, and we also have 2 signals.
+	if(heal_count)
+		heal_count--
+		return
+	heal_count++
+	if(source == link_target)
+		heal_owner = link_target
+		heal_target = link_owner
+	else
+		heal_owner = link_owner
+		heal_target = link_target
+
+	new /obj/effect/temp_visual/telekinesis(get_turf(heal_target))
+	var/heal_amount = round(clamp(amount * (DRONE_ESSENCE_LINK_SHARED_HEAL * stacks), -heal_target.maxHealth, 0))
+	var/heal_remainder = round(max(0, heal_amount - heal_target.getBruteLoss()))
+	heal_target.adjustBruteLoss(heal_amount)
+	heal_target.adjustFireLoss(-heal_remainder)
+	heal_target.adjust_sunder(heal_amount/20)
+	if(heal_amount < 0 || heal_remainder < 0)
+		heal_target.visible_message(span_xenowarning("[heal_target]'s wounds are mended by faint energies."), \
+			span_xenonotice("Through the essence link, [heal_owner] has shared [abs(heal_amount + heal_remainder)] health restoration."))
```

Comment:
I doubt this formula is correct ? You healed between heal amount and zero

Reply:
That formula should be correct. Dividing values such as 20 and 40 would restore 1 and 2 sunder, respectively.

Reply:
Formula was indeed sorta fucked, should be corrected now.